### PR TITLE
Support string path as HDKey.getAccountSigner param

### PR DESCRIPTION
**Pull request by lukaw3d**
Nicer API for https://github.com/oasisprotocol/oasis-wallet-web/pull/990#discussion_r974774873

### CODE REVIEW

**Review by kostko — APPROVED** [review]

File: client-sdk/ts-web/core/src/hdkey.ts [hunk 1]
```diff
@@ -54,10 +50,15 @@ export class HDKey {
     /**
      * Returns the HDKey for the given derivation path
      * using SLIP-0010
-     * @param path Derivation path, starting with m/
+     * @param index Account index
      * @returns Instance of HDKey
      */
-    private derivePath(path: string): HDKey {
+    private derivePath(index: number): HDKey {
```
Comment (pro-wh) [inline client-sdk/ts-web/core/src/hdkey.ts]: should rename this to indicate it no longer takes the whole path

Comment (pro-wh) [reply client-sdk/ts-web/core/src/hdkey.ts]: nvm this is further reworked in the next commit. gonna comment there

File: client-sdk/ts-web/core/src/hdkey.ts [hunk 2]
```diff
@@ -54,10 +50,21 @@ export class HDKey {
     /**
      * Returns the HDKey for the given derivation path
      * using SLIP-0010
-     * @param path Derivation path, starting with m/
+     * @param indexOrPath Account index or derivation path
      * @returns Instance of HDKey
      */
-    private derivePath(path: string): HDKey {
+    private derivePath(indexOrPath: number | string): HDKey {
```
Comment (pro-wh) [inline client-sdk/ts-web/core/src/hdkey.ts:56]: oh ok, then we switch on type inside

**Comment by kostko** [comment]
What's the status of this PR?

**Comment by lukaw3d** [comment]
Abandoned after no follow up in https://github.com/oasisprotocol/oasis-wallet-web/pull/990#discussion_r975440136

**Pull request closed without merging**